app/assets/javascripts/hyrax/collection_types.es6

```diff
@@ -1,94 +1,124 @@
-export default class CollectionTypes {
+import CollectionUtilities from 'hyrax/collections_utils';
 
+export default class CollectionTypes {
   constructor(element) {
+    this.collectionUtilities = new CollectionUtilities();
+
     if (element.length > 0) {
-      this.handleCollapseToggle()
-      this.handleDelete()
+      this.handleCollapseToggle();
+      this.handleDelete();
 
       // Edit Collection Type
-      this.handleAddParticipants()
+      this.setupAddParticipantsHandler();
+      this.participantsAddButtonDisabler();
     }
   }
 
-  handleAddParticipants() {
-    $('#participants').find('.add-participants-form input[type="submit"]').on('click', function(e) {
-      e.preventDefault();
-      const $wrapEl = $(e.target).parents('.form-add-participants-wrapper');
-      if ($wrapEl.length === 0) {
-        return;
-      }
-      const serialized = $wrapEl.find(':input').serialize();
-      const url = '/admin/collection_type_participants?locale=en';
-      if (serialized.length === 0) {
-        return;
-      }
-
-      $.ajax({
-        type: 'POST',
-        url: url,
-        data: serialized
-      }).done(function(response) {
-        // Success handler here, possibly show alert success if page didn't reload?
-      }).fail(function(err) {
-        console.error(err);
-      });
+  setupAddParticipantsHandler() {
+    const { addParticipants } = this.collectionUtilities;
+    const wrapEl = '.form-add-participants-wrapper';
+    const url = '/admin/collection_type_participants?locale=en';
 
-    });
+    $('#participants')
+      .find('.add-participants-form input[type="submit"]')
+      .on(
+        'click',
+        {
+          wrapEl,
+          // This is a callback (seems odd here because just passing in a string value),
+          // because other urls need to be calculated with an id or other param only truly
+          // known from when we know the clicked element's place in DOM.
+          urlFn: e => url
+        },
+        addParticipants.handleAddParticipants.bind(addParticipants)
+      );
   }
 
   handleCollapseToggle() {
-    let $collapseHeader = $('a.collapse-header')
-    let $collapseHeaderSpan = $('a.collapse-header').find('span')
+    let $collapseHeader = $('a.collapse-header');
+    let $collapseHeaderSpan = $('a.collapse-header').find('span');
 
     // Toggle show/hide of collapsible content on bootstrap toggle events
     $('#collapseAbout').on('show.bs.collapse', () => {
-      $collapseHeader.addClass('open')
-      $collapseHeaderSpan.html('Less')
-    })
+      $collapseHeader.addClass('open');
+      $collapseHeaderSpan.html('Less');
```

Comment:
Should the text 'Less/More' be in the translation file?

Reply:
Yea, good call.  Updating now.

Reply:
Actually @lfarrell , I'm not sure the best way to access that rails method `t()` in a javascript file.  Any ideas?

Reply:
I you'll have to have the file name like so `collection_types.es6.erb`. This should work for ES5 files e.g. `collection_types.js.erb`. Not sure what it'll do to es6 files.